### PR TITLE
Rework: Changed Product Model

### PR DESCRIPTION
**Pull request by Magali-D**
# DONE 

- Changed OutOfStock from string to boolean -> error, no doc modification as it was what was expected
- Changed IsHighlanderOfTheMoment: bool to HighlanderOrder: uint with default = 0 -> indicates if the product is Highlander and its order at the same time

### CODE REVIEW

**Review by cedricgautier — APPROVED** [review]

File: docs/database/readme.md [hunk 1]
```diff
@@ -171,7 +170,7 @@ If it's an order it means that it has been paid and that it will be delivered.
 Can be both shipping or billing address.
 The `comment` is optional and can be used to give more information about the address for the delivery person.
 The `zipCode` is a string because some countries have letters in their zipCode.
-The `isActive` field is for history purpose. When an address is updated a new row should be created in the database. 
+The `isActive` field is for history purpose. When an address is updated a new row should be created in the database.
```
Comment (Magali-D) [inline docs/database/readme.md]:
```suggestion
The `isActive` field is for history purpose. When an address is updated a new row should be created in the database. 
```

File: back/src/models/Product.go [hunk 2]
```diff
@@ -12,9 +12,9 @@ type Product struct {
 	PriceWithTaxes          int            `json:"priceWithTaxes"`
 	Description             string         `json:"description"`
 	ThumbnailURL            string         `json:"thumbnailURL"`
-	OutOfStock              string         `json:"outOfStock"              gorm:"default:false"`
+	OutOfStock              bool         	 `json:"outOfStock"              gorm:"default:false"`
 	Materials               []Material     `json:"materials"               gorm:"many2many:materials_of_products"`
-	IsHighlanderOfTheMoment bool           `json:"isHighlanderOfTheMoment" gorm:"default:false"`
+	HighlanderOrder 				uint           `json:"highlanderOrder"         gorm:"default:0"`
```
Comment (barnabepilliaudin) [inline back/src/models/Product.go:17]: @cedricgautier is it a situation where uint can be an issue ?

**Review by BarnabePILLIAUDIN — COMMENTED** [review]
It looks good just need @cedricgautier answer for uint (I don't think it will be an issue but just to be sure an understand why)

File: back/src/models/Product.go (hunk 2 above)
Comment (cedricgautier) [reply back/src/models/Product.go:17]: It should be fine. I don't see any cases where we'll need a negative integer.

Comment (barnabepilliaudin) [reply back/src/models/Product.go:17]: Ok in next sprint we will have to think about it when we will do the validators

**Review by BarnabePILLIAUDIN — APPROVED** [review]

**Pull request merged**